src/crystal/system/win32/visual_studio.cr

```diff
@@ -0,0 +1,55 @@
+require "json"
+
+module Crystal::System::VisualStudio
+  struct Installation
+    include JSON::Serializable
+
+    @[JSON::Field(key: "installationPath")]
+    getter directory : String
+
+    @[JSON::Field(key: "installationVersion")]
+    getter version : String
+
+    # unused fields not mapped
+  end
+
+  def self.find_latest_msvc_path : String?
+    # ported from https://github.com/microsoft/vswhere/wiki/Find-VC
```

Comment:
So need to add a copyright notice?
https://github.com/microsoft/vswhere/wiki/Find-VC#wiki-footer